pkg/controller/control.go

```diff
@@ -323,15 +329,14 @@ func (c *Controller) addReplicaNoLock(newBackend types.Backend, address string,
 		if remain, err := c.backend.RemainSnapshots(); err != nil {
 			return err
 		} else if remain <= 0 {
-			return fmt.Errorf("too many snapshots created")
+			err = fmt.Errorf("too many snapshots created")
```

Comment:
IMO, the error messge should be from `RemainSnapshots`'s `err`. Using `remain <= 0` to chec the `too many snapshots created` eror is a bit of weird.

Reply:
`RemainSnapshots()` only returns numbers of snapshot if the replica is in proper states. 
`RemainSnapshots` would be `maximumChainLength - len(r.activeDiskData)`
So maybe we check `remain <= 0` in `RemainSnapshots()`?
Does anyone know why `maximumChainLength` is 250?

Reply:
I agree with Derek, because at the end you still need to handle `remain <= 0` as an error case, so the error should be good to cover in the `RemainSnapshots` method. However, this is about refactoring, can create a ticket to do it later.

About 250, no ideas, but I believe it's quite a big number for users and it will hurt user experience when the volume used by write heavy application.